src/lib.rs

```diff
@@ -516,11 +516,16 @@ impl<A: Array> ArrayVec<A> {
 
     /// Set the vector’s length without dropping or moving out elements
     ///
-    /// This method is `unsafe` because it changes the notion of the
-    /// number of “valid” elements in the vector. Use with care.
     ///
     /// This method uses *debug assertions* to check that `length` is
     /// not greater than the capacity.
+    ///
+    /// # Safety
+    ///
+    /// This method is `unsafe` because it changes the notion of the
+    /// number of “valid” elements in the vector. This new value
+    /// might not correspond to the actual number of elements in the
+    /// vector which many functions rely to be true. Use with care.
```

Comment:
I wasn't able to just merge this since the way it's written here makes it sound as though it's ok to set the length to a value that does not correspond to the number of initialized elements in the vector. That's not really true: length must be set to a correct value, but the caller is responsible for checking the value (thus the method is unsafe).